src/main/java/com/google/summit/ast/declaration/FieldDeclaration.kt

```diff
@@ -18,24 +18,47 @@ package com.google.summit.ast.declaration
 
 import com.google.summit.ast.Identifier
 import com.google.summit.ast.Node
+import com.google.summit.ast.NodeWithSourceLocation
 import com.google.summit.ast.SourceLocation
 import com.google.summit.ast.TypeRef
 import com.google.summit.ast.expression.Expression
+import com.google.summit.ast.modifier.Modifiable
+import com.google.summit.ast.modifier.Modifier
+
+/**
+ * A group of comma-separated [FieldDeclaration]s that share the same [type] and [modifiers].
+ *
+ * @property type a reference to the type of the field
+ * @param loc the location in the source file
+ */
+class FieldDeclarationGroup(val type: TypeRef, loc: SourceLocation) :
+  NodeWithSourceLocation(loc), Modifiable {
+
+  /** The fields declared in this group. */
+  lateinit var declarations: List<FieldDeclaration>
+
+  override var modifiers: List<Modifier> = emptyList()
+
+  override fun getChildren(): List<Node> = modifiers + type + declarations
+}
 
 /**
  * A declaration for a field (a.k.a. class member).
  *
  * @param id the name of the field
- * @property type a reference to the type of the field
  * @property initializer an optional initializer expression
- * @param sourceLocation the location in the source file
+ * @param group the [FieldDeclarationGroup] that contains this field
+ * @param loc the location in the source file
  */
 class FieldDeclaration(
   id: Identifier,
-  val type: TypeRef,
   val initializer: Expression?,
+  group: FieldDeclarationGroup,
   loc: SourceLocation
-) : Declaration(id, loc) {
+) : Declaration(id, loc), Modifiable by group {
+
+  override fun getChildren(): List<Node> = listOfNotNull(id, initializer)
 
-  override fun getChildren(): List<Node> = modifiers + listOfNotNull(id, type, initializer)
+  override fun getEnclosingType(): TypeDeclaration? =
+    (parent as FieldDeclarationGroup).parent as? TypeDeclaration
```

Comment:
For all other `Declaration`s, the direct `parent` is a `TypeDeclaration` and thus the enclosing type. This is not the case for `FieldDeclaration`s anymore. Is there a better way to structure this than overriding `getEnclosingType`?

Reply:
One application for the chain of enclosing types is to get a sequence of identifiers when constructing a qualified name.

Here it seems appropriate to skip over the `FieldDeclarationGroup`, which doesn't have an identifier.

I don't have an opinion here, if you think it is cleaner, you could also implement a `FieldDeclarationGroup.getEnclosingType` method as a helper, without it implementing any interface.